model/src/gateway/presence/activity_button.rs

```diff
@@ -209,6 +212,9 @@ impl Serialize for ActivityButton {
                 state.end()
             }
             Self::Text(text) => serializer.serialize_str(&text.label),
+            ActivityButton::Unknown => Err(serde::ser::Error::custom(
```

Comment:
we should use `Self` here, also can we find a way to remove the FQN for `serde::ser::Error`, maybe `use serde::ser::Error as SerError`?
```suggestion
            Self::Unknown => Err(serde::ser::Error::custom(
```